### PR TITLE
Opsgenie Alert Provider

### PR DESCRIPTION
**Pull request by vinicius73**
## Opsgenie Alert Provider

New alert provider with [Opsgenie](https://www.atlassian.com/software/opsgenie)
Ref: https://docs.opsgenie.com/docs/alert-api

It turns Gatus able to open and close alerts in [Opsgenie API](https://docs.opsgenie.com/docs/alert-api)

#### Config

```yml
opsgenie:
    api-key: api-key
    priority: P1
    source: gatus
    entity-prefix: gatus-
    alias-prefix: gatus-healthcheck-
    tags:
      - my:extra:tag
````

#### Todo

- [x] Implementation
- [x] Tests
- [x] Doc

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -311,8 +311,10 @@ ignored.
 | Parameter              | Description                                                                                                            | Default        |
 |:-----------------------|:---------------------------------------------------------------------------------------------------------------------- |:-------|
 | `alerting.discord`     | Configuration for alerts of type `discord`. <br />See [Configuring Discord alerts](#configuring-discord-alerts).             | `{}`   |
-| `alerting.mattermost`  | Configuration for alerts of type `mattermost`. <br />See [Configuring Mattermost alerts](#configuring-mattermost-alerts).   | `{}`   |
+| `alerting.email`       | Configuration for alerts of type `email`. <br />See [Configuring Email alerts](#configuring-email-alerts).                   | `{}`   |
```
Comment (vinicius73) [inline README.md:314]: That was missing in this table.

File: alerting/provider/opsgenie/opsgenie.go [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+	if endpoint.Group != "" {
+		message = fmt.Sprintf("[%s] %s", endpoint.Group, message)
+	}
+
+	for _, conditionResult := range result.ConditionResults {
+		var prefix string
+		if conditionResult.Success {
+			prefix = "▣"
+		} else {
+			prefix = "▢"
```
Comment (TwiN) [inline alerting/provider/opsgenie/opsgenie.go:166]: oO I see you're going with different characters for the success/failure.

Well, that's fine with me. Your implementation, your choice 🤣

File: alerting/provider/opsgenie/opsgenie_test.go [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+	"reflect"
+	"testing"
+)
+
+func TestAlertProvider_IsValid(t *testing.T) {
+	invalidProvider := AlertProvider{APIKey: ""}
+	if invalidProvider.IsValid() {
+		t.Error("provider shouldn't have been valid")
+	}
+	validProvider := AlertProvider{APIKey: "2f8fe136-27a6-4a79-ab5b-a13cb0e59699"}
```
Comment (TwiN) [inline alerting/provider/opsgenie/opsgenie_test.go:18]: Just making sure, this isn't your production API key, right? 🤣

**Review by TwiN — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (TwiN) [reply README.md:314]: Good catch!

File: alerting/alert/type.go [hunk 4]
```diff
@@ -34,4 +34,7 @@ const (
 
 	// TypeTwilio is the Type for the twilio alerting provider
 	TypeTwilio Type = "twilio"
+
+	// Opsgenie is the Type for the opsgenie alerting provider
+	Opsgenie Type = "opsgenie"
```
[the remark this answers is not in the document]
Comment (vinicius73) [reply alerting/alert/type.go:39]: No 🤣
It's a random uuid, I will put a placeholder to keep it clean 👍🏼

File: alerting/provider/opsgenie/opsgenie.go (hunk 2 above)
Comment (vinicius73) [reply alerting/provider/opsgenie/opsgenie.go:166]: I was trying simulate slack messages 🤓

File: alerting/provider/opsgenie/opsgenie_test.go (hunk 3 above)
Comment (vinicius73) [reply alerting/provider/opsgenie/opsgenie_test.go:18]: No 🤣
It's a random uuid, I will put a placeholder to keep it clean 👍🏼

**Comment by TwiN** [comment]
Awesome work! Thank you for your contribution @vinicius73

**Comment by vinicius73** [comment]
Thank for the great project @TwiN .
Can you create a new gatus docker image with this update?

**Comment by TwiN** [comment]
@vinicius73 Pleasure's mine!

It's already available under `twinproduction/gatus:latest`.

Every push to master updates the `latest` tag; if it's not working on your end, it's probably because you have the old `latest` tag cached on your machine.

**Comment by vinicius73** [comment]
LOL :raised_hands: makes sense @TwiN :joy: 
Thanks again

**Comment by TwiN** [comment]
Released in [v3.5.0](https://github.com/TwiN/gatus/releases/tag/v3.5.0)

**Pull request merged**